mllib/src/main/scala/org/apache/spark/ml/param/shared/sharedGeneralTypeParams.scala

```diff
@@ -0,0 +1,34 @@
+/*
```

Comment:
I think the filename should be `GenericTypeParams`, right? Will be nice to integrate into the other part of `Params`. Maybe we can use scala macro?